src/conn.rs

```diff
@@ -202,10 +207,11 @@ pub struct InProgress<'a, 'c> {
     generation: u64,
     partition_map: PartitionMap,
     schema: Schema,
-
     cache: InProgressSQLiteAttributeCache,
-
     use_caching: bool,
+    // TODO: Collect txids/affected datoms in a better way
+    tx_reports: SmallVec<[TxReport; 4]>,
```

Comment:
This, btw, is one reason to keep `TxReport` small: the purpose of `smallvec` is that it can often stack-allocate, and so here we are increasing the size of `InProgress` by approximately `4 * sizeof(TxReport)`.